File: backend/app/schemas/interaction.py
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.models.interaction import InteractionType


class InteractionCreate(BaseModel):
    hcp_id: str
    interaction_type: InteractionType
    raw_input: str
    notes: Optional[str] = None


class InteractionUpdate(BaseModel):
    notes: Optional[str] = None


class InteractionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID

    hcp_id: str

    interaction_type: InteractionType

    raw_input: str

    notes: Optional[str] = None

    ai_summary: Optional[str] = None

    sentiment: Optional[str] = None

    compliance_status: Optional[str] = None

    follow_up: Optional[str] = None

    priority: Optional[str] = None

    interaction_date: datetime

    created_at: datetime

    updated_at: datetime

from app.ai.models import AIAnalysis


class ProcessInteractionResponse(BaseModel):
    interaction: InteractionResponse
    analysis: AIAnalysis

File: frontend/src/types/interaction.ts
export interface Interaction {
  id: string;
  hcp_id: string;

  interaction_type: string;

  raw_input: string;

  notes: string | null;

  ai_summary: string;

  sentiment: string;

  compliance_status: string;

  follow_up: string;

  priority: string;

  interaction_date: string;

  created_at: string;

  updated_at: string;
}

export interface CreateInteractionRequest {
  hcp_id: string;
  interaction_type: string;
  raw_input: string;
  notes?: string | null;
}

export interface UpdateInteractionRequest {
  interaction_type: string;
  raw_input: string;
  notes?: string | null;
}